yarn-project/sequencer-client/src/publisher/l1-publisher.ts

```diff
@@ -416,45 +450,38 @@ export class L1Publisher {
       return true;
     }
 
-    this.metrics.recordFailedTx('process');
-
-    const errorMsg = await this.tryGetErrorFromRevertedTx({
-      args,
-      functionName,
-      gasLimit,
-      abi: RollupAbi,
-      address: this.rollupContract.address,
-      blockNumber: receipt.blockNumber,
-    });
-    this.log.error(`Rollup process tx reverted. ${errorMsg}`, undefined, {
-      ...ctx,
-      txHash: receipt.transactionHash,
-    });
     await this.sleepOrInterrupted();
     return false;
   }
 
-  private async tryGetErrorFromRevertedTx(args: {
-    args: any[];
-    functionName: string;
-    gasLimit: bigint;
-    abi: any;
-    address: Hex;
-    blockNumber: bigint | undefined;
-  }) {
-    try {
-      await this.publicClient.simulateContract({ ...args, account: this.walletClient.account });
-      return undefined;
-    } catch (err: any) {
-      if (err.name === 'ContractFunctionExecutionError') {
-        const execErr = err as ContractFunctionExecutionError;
-        return compactArray([
-          execErr.shortMessage,
-          ...(execErr.metaMessages ?? []).slice(0, 2).map(s => s.trim()),
-        ]).join(' ');
-      }
-      this.log.error(`Error getting error from simulation`, err);
+  private tryGetErrorFromRevertedTx(
+    err: any,
+    args: {
+      args: any[];
+      functionName: string;
+      abi: any;
+      address: Hex;
+    },
+  ) {
+    // We can no longer simulate contract calls with blobs, as there is no way to mock the blob hash checked in the contract
+    // The simulation will fail with 'incorrect blob hash' regardless of the real reason
+    // Instead, we pass the sendTransaction error here, and use viem to convert it to a ContractFunctionExecutionError
+    // (The actual error is a TransactionExecutionError, which does not display our custom errors)
+    const contractErr =
+      err.name === 'ContractFunctionExecutionError'
+        ? err
+        : getContractError(err as BaseError, {
+            ...args,
+            sender: this.account.address,
+            docsPath: '/docs/contract/simulateContract',
+          });
+    if (contractErr.name === 'ContractFunctionExecutionError') {
+      const execErr = contractErr as ContractFunctionExecutionError;
+      return compactArray([execErr.shortMessage, ...(execErr.metaMessages ?? []).slice(0, 2).map(s => s.trim())]).join(
+        ' ',
+      );
```

Comment:
Sadly had to butcher @spalladino 's nice error handling, as we no longer get `ContractFunctionExecutionError`s from `propose` transactions - these come from `contract.write` functions which do not accept blob transactions at the moment. Instead, I have to use `client.sendTransaction`, and use a bit of a hack to get it to return our custom error name.

Reply:
> We can no longer simulate contract calls with blobs, as there is no way to mock the blob hash checked in the contract

This sounds bad. Does this mean we need to just send the tx without being able to simulate it first? Should we tweak the structure of the contract so we _can_ mock the blob hash in a simulation, perhaps abusing state overrides?

Reply:
I was thinking about having a "check blobs" flag always set to true, but we set it to false via state overrides during simulation. But that's a bit shitty in that we always pay an extra SLOAD just to do simulations.

Reply:
That sounds like the nicest interim we can have atm! Happy to implement it and revert to the old simulation